File: Task_2/src/cartoon.cpp
#include <string>
#include "libfilm/cartoon.hpp"

Cartoon::Cartoon(std::string name, uint32_t release_year, Type type) :
    Film(std::move(name), release_year), m_type(type) {}

void Cartoon::PrintDetailsToOStream(std::ostream& output_stream) const
{
	output_stream << "Film type: cartoon\n";
	output_stream << "Cartoon type: " << CartoonTypeToString(m_type) << '\n';
}

const char* Cartoon::CartoonTypeToString(Type type)
{
	static const char kUndefined[] = "undefined";
	static const char kPuppet[] = "puppet";
	static const char kClay[] = "clay";
	static const char kHanddrawn[] = "handdrawn";
	switch (type)
	{
	case Type::kPuppetCartoon:
		return kPuppet;
	case Type::kClayAnimation:
		return kClay;
	case Type::kHandDrawnCartoon:
		return kHanddrawn;
	default:
		return kUndefined;
	}
}

File: Task_3/libfilm/cartoon.py
from enum import Enum

from libfilm.film import Film


class Cartoon(Film):
    class CartoonType(Enum):
        PUPPET = 'puppet'
        CLAY = 'clay'
        HANDDRAWN = 'handdrawn'

        def __str__(self):
            return str(self.value)

    def __init__(self, name: str, release_year: int, cartoon_type: CartoonType):
        super().__init__(name, release_year)
        self._cartoon_type = cartoon_type

    @property
    def type(self):
        return self._cartoon_type
